Services/WebToken/AccessTokenLoader.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+namespace Lexik\Bundle\JWTAuthenticationBundle\Services\WebToken;
+
+use Jose\Bundle\JoseFramework\Services\ClaimCheckerManager;
+use Jose\Bundle\JoseFramework\Services\ClaimCheckerManagerFactory;
+use Jose\Bundle\JoseFramework\Services\HeaderCheckerManager;
+use Jose\Bundle\JoseFramework\Services\JWELoader;
+use Jose\Bundle\JoseFramework\Services\JWELoaderFactory;
+use Jose\Bundle\JoseFramework\Services\JWSLoader;
+use Jose\Bundle\JoseFramework\Services\JWSLoaderFactory;
+use Jose\Component\Checker\InvalidClaimException;
+use Jose\Component\Checker\MissingMandatoryClaimException;
+use Jose\Component\Core\JWKSet;
+use Lexik\Bundle\JWTAuthenticationBundle\Exception\JWTDecodeFailureException;
+
+final class AccessTokenLoader
+{
+    /**
+     * @var JWSLoader
+     */
+    private $jwsLoader;
+
+    /**
+     * @var HeaderCheckerManager
+     */
+    private $jwsHeaderCheckerManager;
+
+    /**
+     * @var ClaimCheckerManager
+     */
+    private $claimCheckerManager;
+
+    /**
+     * @var null|JWELoader
+     */
+    private $jweLoader = null;
+
+    /**
+     * @var JWKSet
+     */
+    private $signatureKeyset;
+
+    /**
+     * @var JWKSet|null
+     */
+    private $encryptionKeyset;
+
+    /**
+     * @var string[]
+     */
+    private $mandatoryClaims;
+
+    /**
+     * @var bool|null
+     */
```

Comment:
let's rely on IDEs inferring the type from constructor params and remove property types (sorry if you did this for consistency, the whole codebase will be refactored at some point)

Reply:
Sure. No pb

Reply:
Done